core: `provider` block inside `resource` causes crash on `plan`

I've just downloaded and started playing around with Terraform. Creating a simple `aws.tf`config file, then running `terraform plan`, gave a ***panic: runtime error: index out of range*** .

Here's the version.
```
$ terraform --version
Terraform v0.6.13
```

Here's the (probably erroneous) config file.
```
$ cat aws.tf 
resource "aws" "web" {
  provider = "aws" {
    region = "us-west-2"
  }
}
```

And here's the crash log
```
$  terraform plan
panic: runtime error: index out of range

goroutine 1 [running]:
panic(0xa03460, 0x82131c080)
	/opt/go/src/runtime/panic.go:464 +0x3e6
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decodeMap(0x82143ca58, 0xb8dda8, 0x4, 0x88213604c0, 0x82144bca0, 0x821480, 0x82133a1d0, 0x195, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:330 +0x1363
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.(*decoder).decode(0x82143ca58, 0xb8dda8, 0x4, 0x88213604f0, 0x821325560, 0x821480, 0x82133a1d0, 0x195, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:89 +0x7f4
github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl.DecodeObject(0x7ecc60, 0x82133a1d0, 0x88213604f0, 0x821325560, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/hashicorp/hcl/decoder.go:49 +0x25f
github.com/hashicorp/terraform/config.loadResourcesHcl(0x82144be00, 0x0, 0x0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/loader_hcl.go:441 +0x939
github.com/hashicorp/terraform/config.(*hclConfigurable).Config(0x82144bce0, 0xb85318, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/loader_hcl.go:118 +0xd66
github.com/hashicorp/terraform/config.(*importTree).ConfigTree(0x8214a6b00, 0x23, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/import_tree.go:87 +0x68
github.com/hashicorp/terraform/config.LoadFile(0x8214a0fc0, 0x23, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/loader.go:44 +0x80
github.com/hashicorp/terraform/config.LoadDir(0x821314124, 0x1c, 0x11, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/loader.go:91 +0x38c
github.com/hashicorp/terraform/config/module.NewTreeModule(0x0, 0x0, 0x821314124, 0x1c, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/config/module/tree.go:40 +0x41
github.com/hashicorp/terraform/command.(*Meta).Context(0x82149a0f0, 0x821314124, 0x1c, 0xc577e0, 0x11, 0x0, 0xa, 0x4, 0x7fff5fbff60e, 0x0, ...)
	/opt/gopath/src/github.com/hashicorp/terraform/command/meta.go:148 +0x509
github.com/hashicorp/terraform/command.(*PlanCommand).Run(0x82149a0f0, 0x8213142b0, 0x0, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/command/plan.go:66 +0x80d
github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli.(*CLI).Run(0x8213c4600, 0x8214a0600, 0x0, 0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/vendor/github.com/mitchellh/cli/cli.go:153 +0x5ee
main.wrappedMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:140 +0xb05
main.realMain(0x0)
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:76 +0xa10
main.main()
	/opt/gopath/src/github.com/hashicorp/terraform/main.go:19 +0x18



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

--- a/Godeps/Godeps.json
+++ b/Godeps/Godeps.json
@@ -653,39 +653,39 @@
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hcl",
-			"Rev": "71c7409f1abba841e528a80556ed2c67671744c3"
+			"Rev": "2604f3bda7e8960c1be1063709e7d7f0765048d0"
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hcl/hcl/ast",
-			"Rev": "71c7409f1abba841e528a80556ed2c67671744c3"
+			"Rev": "2604f3bda7e8960c1be1063709e7d7f0765048d0"
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hcl/hcl/parser",
-			"Rev": "71c7409f1abba841e528a80556ed2c67671744c3"
+			"Rev": "2604f3bda7e8960c1be1063709e7d7f0765048d0"
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hcl/hcl/scanner",
-			"Rev": "71c7409f1abba841e528a80556ed2c67671744c3"
+			"Rev": "2604f3bda7e8960c1be1063709e7d7f0765048d0"
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hcl/hcl/strconv",
-			"Rev": "71c7409f1abba841e528a80556ed2c67671744c3"
+			"Rev": "2604f3bda7e8960c1be1063709e7d7f0765048d0"
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hcl/hcl/token",
-			"Rev": "71c7409f1abba841e528a80556ed2c67671744c3"
+			"Rev": "2604f3bda7e8960c1be1063709e7d7f0765048d0"
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hcl/json/parser",
-			"Rev": "71c7409f1abba841e528a80556ed2c67671744c3"
+			"Rev": "2604f3bda7e8960c1be1063709e7d7f0765048d0"
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hcl/json/scanner",
-			"Rev": "71c7409f1abba841e528a80556ed2c67671744c3"
+			"Rev": "2604f3bda7e8960c1be1063709e7d7f0765048d0"
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hcl/json/token",
-			"Rev": "71c7409f1abba841e528a80556ed2c67671744c3"
+			"Rev": "2604f3bda7e8960c1be1063709e7d7f0765048d0"
 		},
 		{
 			"ImportPath": "github.com/hashicorp/hil",

